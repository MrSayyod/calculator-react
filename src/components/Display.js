import PropTypes from 'prop-types'

class Display extends React.Component {
  constructor(props) {
    super(props)
  }

  render() {
    return (
      <div>
        <h1>{props.result}</h1>
      </div>
    )
  }
}

Display.propTypes = {
  result: PropTypes.string.isRequired
}

Display.defaultProps = {
  result: '0'
}