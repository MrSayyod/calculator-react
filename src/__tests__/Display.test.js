import React from 'react';
import { render, screen } from '@testing-library/react';
import { unmountComponentAtNode } from 'react-dom';
import '@testing-library/jest-dom/extend-expect';
import Display from '../components/Display';

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('displays the result', () => {
  const { renderer } = render(<Display total="1" />);
  expect(screen.getByTestId('display-test')).toHaveTextContent('1');
  expect(renderer).toMatchSnapshot();
});
