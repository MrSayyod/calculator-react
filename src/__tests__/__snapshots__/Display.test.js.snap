// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays the result 1`] = `undefined`;
